test/src/cross_currencies.test.js

```diff
@@ -202,6 +202,50 @@ nano_environments.forEach(function(model) {
 });
 
 
+nano_environments.forEach(function(model) {
+    test(`[Nano ${model.letter}] Swap from AE ERC20 to ETH`, zemu(model, async (sim) => {
+        let transaction = new SwapTransactionPerformer(model, sim);
+        transaction.setFromCurrencyInfo(AE_INFO);
+        transaction.setToCurrencyInfo(ETH_INFO);
+        // 1.1234 AE to 1 BTC
+        transaction.setAmountToProvider(1000000 * 1000000 * 1000000 * 1.1234); // 10^18 wei == 1 ETH
+        transaction.setAmountToWallet(100000000);
+        transaction.setFee(1477845000000000);
+        await transaction.performSuccessfulTransaction();
+
+        let transport = await sim.getTransport();
+
+        const eth = new Eth(transport);
+        const aeInfo = byContractAddress("0x5CA9a71B1d01849C0a95490Cc00559717fCF0D1d")
+        if (aeInfo) await eth.provideERC20TokenInformation(aeInfo)
+
+        await expect(eth.signTransaction("44'/60'/0'/0/0", 'F8690385098BCA5A00828CCD945CA9a71B1d01849C0a95490Cc00559717fCF0D1d80B844A9059CBB000000000000000000000000d692Cb1346262F584D17B4B470954501f6715a820000000000000000000000000000000000000000000000000F971E5914AC8000038080'))
+            .resolves.toEqual({
+                "r": "6e766ca0c8474da1dc5dc0d057e0f97711fd70aed7cb9965ff6dc423d8f4daad",
+                "s": "63a0893b73e752965b65ebe13e1be8b5838e2113006656ea2eefa55fe0fa2919",
+                "v": "2a"
+            });
+    }))
+});
+
+
+nano_environments.forEach(function(model) {
+    test(`[Nano ${model.letter}] Swap from ETH to SHIB ERC20, large amount does not overflow`, zemu(model, async (sim) => {
+        let transaction = new SwapTransactionPerformer(model, sim);
+        transaction.setFromCurrencyInfo(ETH_INFO);
+        transaction.setToCurrencyInfo(SHIB_INFO);
+        // 1.1234 AE to 1 BTC
+        transaction.setAmountToProvider(1000000 * 1000000 * 1000000 * 0.3); // 10^18 wei == 1 ETH
+        transaction.setAmountToWallet(1000000 * 1000000 * 1000000 * 2318520.24);
+        transaction.setFee(1477845000000000);
+        let right_clicks = (model.letter ==  'X' ? 4 : 6);
+        await transaction.performSuccessfulTransaction(right_clicks);
+
+        let transport = await sim.getTransport();
```

Comment:
all `let` can be `const`.

Is this last line doing anything? If no, remove, otherwise it can just keep it without the assignment.